### PR TITLE
Add overloads for NamingProvider.

### PR DESCRIPTION
**Pull request by kimlundjohansen**

### CODE REVIEW

File: src/Atc.Azure.Options/Extensions/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -10,6 +10,9 @@ namespace Atc.Azure.Options.Extensions
 {
     public static class ConfigurationExtensions
     {
+        public static void ConfigureKeyVaultForApi(this IConfigurationBuilder config)
+            => ConfigureKeyVaultForApi(config, new NamingProvider());
```
Comment (cjakobsen) [inline src/Atc.Azure.Options/Extensions/ConfigurationExtensions.cs:14]: Directly new'ing this up here might make it harder to use a custom NamingProvider.

Comment (kimlundjohansen) [reply src/Atc.Azure.Options/Extensions/ConfigurationExtensions.cs:14]: There is an overload where you can specify a different NamingProvider. Is that not sufficient?

Comment (lupusbytes) [reply src/Atc.Azure.Options/Extensions/ConfigurationExtensions.cs:14]: This is just an overload. 
There is still the:
```
ConfigureKeyVaultForApi(
            this IConfigurationBuilder config,
            INamingProvider namingProvider,
            string environmentOptionsSectionName = nameof(EnvironmentOptions),
            string namingOptionsSectionName = nameof(NamingOptions))
```
Or did i misunderstand something ?

Comment (cjakobsen) [reply src/Atc.Azure.Options/Extensions/ConfigurationExtensions.cs:14]: Yes, missed that

**Review by cjakobsen — APPROVED** [review]

**Pull request merged**